src/Compilers/Core/AnalyzerDriver/AnalyzerManager.cs

```diff
@@ -265,12 +265,6 @@ internal bool IsSupportedDiagnostic(DiagnosticAnalyzer analyzer, Diagnostic diag
 
             foreach (var diag in supportedDiagnostics)
             {
-                if (HasNotConfigurableTag(diag.CustomTags))
```

Comment:
If ruleset has an entry for non-configurable diagnostic which is not Action=None, that enables the diagnostic. Analyzer will always execute and if it reports that diagnostic, compiler will always create it with default severity.

We probably can have a better design if ruleset file just allowed stating IsEnabled flag explicitly. Then we won't have to use Action='something' setting just to turn on diagnostics.